### PR TITLE
Remove stale OPAL dss and opal_tree code

### PR DESCRIPTION
**Pull request by rhc54**
With the switch to PRRTE, there is no remaining need or use
for the dss - it is all in PRRTE now. The only use was to
pack/unpack the opal_tree class, but that class is also
no longer used anywhere.

Signed-off-by: Ralph Castain <rhc@pmix.org>

### CODE REVIEW

**Comment by ibm-ompi** [comment]
The IBM CI (GNU/Scale) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/e5971ff176f9afb944f60a60037ed8fe

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/26fb857e4b8d4a08198da5e876bba283

**Comment by ibm-ompi** [comment]
The IBM CI (PGI) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/da859e5606b8e9827c6d9874c6e83b81

**Comment by ibm-ompi** [comment]
The IBM CI (GNU/Scale) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/61005a91f2b15d2900e28f47d06e5c5d

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/8e48a043ce25108989cc237ce9336f3f

**Comment by ibm-ompi** [comment]
The IBM CI (PGI) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/34058e78bfb5d5a626480eb7e0c7505d

**Comment by ibm-ompi** [comment]
The IBM CI (GNU/Scale) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/70a454418f51968f23319dbe85ec0b9d

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/ea5fd2d68f3dac0156c5a836c03b623e

**Comment by jjhursey** [comment]
The IBM CI is marking as passed, but every unit test is emiting:
```
09:13:56 [89bfb650e73f:00329] PMIX ERROR: OUT-OF-RESOURCE in file base/bfrop_base_copy.c at line 81
09:13:58 [89bfb650e73f:00329] [grpcomm_direct.c:231] PMIx Error: OUT-OF-RESOURCE
```

**Comment by rhc54** [comment]
> The IBM CI is marking as passed, but every unit test is emiting:
>09:13:56 [89bfb650e73f:00329] PMIX ERROR: OUT-OF-RESOURCE in file base/bfrop_base_copy.c at line 81
>09:13:58 [89bfb650e73f:00329] [grpcomm_direct.c:231] PMIx Error: OUT-OF-RESOURCE

Well that's cute! I'll take a look - I would expect that also happened on earlier PRs too as this one didn't change that code area.

**Comment by jjhursey** [comment]
Yeah. The test case didn't generate any other output than those error messages (no application output), and the test case timed out. I'm not sure why CI didn't flag it as a failure - I'll look into that.

**Comment by jjhursey** [comment]
It looks like it was introduced in https://github.com/open-mpi/ompi/pull/8495 - PRs tested before that was merged had correct output, those posted afterward show this behavior. I'm working on fixing CI to catch it.

**Comment by rhc54** [comment]
I found the issue - I'll post the fix separately and let it go thru the modified CI when you are ready.

**Comment by rhc54** [comment]
@jjhursey I added the fix for what you observed into https://github.com/open-mpi/ompi/pull/8497. Could you please re-start the CI on that PR when your fix is ready? Only constraint - I need to get that PR committed prior to MTT tarball generation tonight to fix something borking MTT.

**Comment by jjhursey** [comment]
I can confirm that https://github.com/open-mpi/ompi/pull/8497 fixed the IO issue I mentioned earlier. Thanks!

File: opal/mca/btl/sm/btl_sm_module.c [hunk 1]
```diff
@@ -332,7 +333,6 @@ static int sm_add_procs (struct mca_btl_base_module_t* btl,
                             struct mca_btl_base_endpoint_t **peers,
                             opal_bitmap_t *reachability)
 {
-    mca_btl_sm_component_t *component = &mca_btl_sm_component;
```
Comment (awlauria) [inline opal/mca/btl/sm/btl_sm_module.c:335]: Unused variable warning?

File: ompi/communicator/comm.c [hunk 2]
```diff
@@ -1662,21 +1663,19 @@ int ompi_comm_get_rprocs ( ompi_communicator_t *local_comm,
...
-        goto err_exit;
-    }
+    PMIX_DATA_BUFFER_LOAD(rbuf, recvbuf, rlen);
 
     /* decode the names into a proc-list -- will never add a new proc
        as the result of this operation, so no need to get the newprocs
        list or call PML add_procs(). */
     rc = ompi_proc_unpack(rbuf, rsize, &rprocs, NULL, NULL);
-    OBJ_RELEASE(rbuf);
+    PMIX_DATA_BUFFER_RELEASE(rbuf);
```
Comment (awlauria) [inline ompi/communicator/comm.c:1678]: Does the rc need to be converted in these cases via `opal_pmix_convert_status()`

**Review by awlauria — COMMENTED** [review]
Looks pretty straight forward to me, just the one question.

File: ompi/communicator/comm.c [hunk 3]
```diff
@@ -1579,7 +1579,8 @@ int ompi_comm_get_rprocs ( ompi_communicator_t *local_comm,
         if ( OMPI_SUCCESS != rc ) {
             goto err_exit;
         }
-        if (OPAL_SUCCESS != (rc = opal_dss.unload(sbuf, &sendbuf, &size_len))) {
+        PMIX_DATA_BUFFER_UNLOAD(sbuf, sendbuf, size_len);
+        if (NULL == sendbuf) {
             goto err_exit;
```
Comment (awlauria) [inline ompi/communicator/comm.c:1584]: I think the rc needs to be set here?

File: opal/mca/btl/sm/btl_sm_module.c (hunk 1 above)
Comment (rhc54) [reply opal/mca/btl/sm/btl_sm_module.c:335]: Yeah - I don't think these changes caused it to become unused, but it was simple enough to remove anyway.

**Review by awlauria — APPROVED** [review]
LGTM

**Comment by awlauria** [comment]
@rhc54 it looks like a submodule pointer update may have snuck in here..Is that intentional?

**Comment by rhc54** [comment]
yes - keeping things in sync

**Comment by awlauria** [comment]
Good to go?

**Comment by rhc54** [comment]
Oh - yeah, I just plumb forgot! What a day...

**Pull request merged**